File: KitDemo/ViewController.m
//
//  ViewController.m
//  KitDemo
//
//  Created by Hu on 16/7/29.
//  Copyright © 2016年 redye. All rights reserved.
//

#import "ViewController.h"
#import "CarouselViewController.h"

NSString * const kCell = @"Cell";

@interface ViewController ()<UITableViewDataSource, UITableViewDelegate>
@property (nonatomic, strong) UITableView *tableView;
@property (nonatomic, strong) NSArray *classes;
@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    [self initData];
    
    [self setUI];
}

- (void)initData {
    self.classes = @[[CarouselViewController class]];
}

- (void)setUI {
    self.title = @"Home";
    
    _tableView = [[UITableView alloc] initWithFrame:self.view.bounds style:UITableViewStylePlain];
    _tableView.dataSource = self;
    _tableView.delegate = self;
    [_tableView registerClass:[UITableViewCell class] forCellReuseIdentifier:kCell];
    [self.view addSubview:_tableView];
}

#pragma mark - UITableViewDataSource, UITableViewDelegate
- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    return self.classes.count;
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:kCell forIndexPath:indexPath];
    cell.selectionStyle = UITableViewCellSelectionStyleNone;
    NSString *text = NSStringFromClass(self.classes[indexPath.row]);
    cell.textLabel.text = text;
    return cell;
}

- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath {
    Class class = self.classes[indexPath.row];
    UIViewController *viewController = [class new];
    [self.navigationController pushViewController:viewController animated:YES];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    
    
}

@end

File: KitDemo-Swift/KitDemo-Swift/ViewController.swift
//
//  ViewController.swift
//  KitDemo-Swift
//
//  Created by Hu on 16/8/3.
//  Copyright © 2016年 redye. All rights reserved.
//

import UIKit

let kCell: String = "Cell"

class ViewController: UIViewController, UITableViewDataSource, UITableViewDelegate {

    ///这里是属性
    var classes:[AnyClass] = [AnyClass]()
    var tableView: UITableView!
    
    
    /**
     @brief 这里是方法
     */
    override func viewDidLoad() {
        super.viewDidLoad()
        
        self.initData()
        
        self.setUI()
    }
    
    func initData() {
        classes = [CarouselViewController.self,
                   WelcomeViewController.classForCoder(),
                   RatingViewController.classForCoder(),
                   ScrollControllViewController.classForCoder(),
                   PageControlViewController.classForCoder(),
                   PasswordFieldViewController.classForCoder(),
                   SelectionViewController.classForCoder()]

    }
    
    func setUI() {
        self.title = "Home"
        
        tableView = UITableView(frame: self.view.bounds, style: .plain)
        tableView.register(UITableViewCell.classForCoder(), forCellReuseIdentifier: kCell)
        tableView.dataSource = self
        tableView.delegate = self
        self.view.addSubview(tableView)
    }
    
    // MARK: - UITableViewDatasource, UITableViewDelegate
    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return self.classes.count
    }
    
    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let cell = tableView.dequeueReusableCell(withIdentifier: kCell, for: indexPath)
        cell.selectionStyle = .none
        let text = String(describing: self.classes[(indexPath as NSIndexPath).row])
        cell.textLabel?.text = text
        return cell
    }
    
    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        let cls: AnyClass! = self.classes[(indexPath as NSIndexPath).row]
        let viewController = (cls as! UIViewController.Type).init()
        self.navigationController?.pushViewController(viewController, animated: true)
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

